### DISCUSSION TITLE
Logs from FST WiFi

### DISCUSSION THREAD

**Discussion by RosemaryOrchard** [andrey-yantsen/timeguard-mqtt#14] [General]
I haven't got a whole 24 hours of logs here; apparently, the FSTWiFi switches are _very_ chatty. But I have cleared my logging and tracked two events, switching permanent off to permanent on in the app, and vice versa.

https://gist.github.com/RosemaryOrchard/55925620a7cf43021171e64156f70656

**Comment by worried-networking** [comment]
Wow, awesome! Thank you! I've been waiting for these logs for quite some time :D

How chatty your FSTWiFi is? My NTTWIFI  regularly sends "ping" messages every ≈40 seconds, with a few occasional other messages. 

Judging by the logs, it should be working already. Can you please confirm if you could see the switch's state in MQTT and control it? Do you have Home Assistant installed, or are you planning to use the integration directly via MQTT?

I'd need to see about 5 minutes of the initial device chatter if it is indeed working: I want to see if there's any difference between NTTWiFi and FSTWiFi. Starting from the moment the device boots — just shut off the mains for a few seconds and turn it on again.

**Comment by RosemaryOrchard** [comment]
>  I've been waiting for these logs for quite some time :D 

Haha, that's on me. I ordered it, it took a while to arrive. Then I had to get it installed, and then I had to setup a whole new home network 😂 

> Judging by the logs, it should be working already. Can you please confirm if you could see the switch's state in MQTT and control it? Do you have Home Assistant installed, or are you planning to use the integration directly via MQTT?

It is indeed working! I have HA installed, and set it up to communicate with the MQTT addon, auto-discovery, etc., and once I got the hang of "the people who programmed this before never accounted for someone just wanting an on/off button!" I figured it out :D

> I'd need to see about 5 minutes of the initial device chatter if it is indeed working: I want to see if there's any difference between NTTWiFi and FSTWiFi. Starting from the moment the device boots — just shut off the mains for a few seconds and turn it on again.

Hmmm, flipping the switch on the device itself does not produce any logs (probably as it can still send and receive info then). I'll figure something out that doesn't involve turning most of the house off at the consumer unit!

I've just added the log from the first 5 minutes of the container though: [startup](https://gist.github.com/RosemaryOrchard/55925620a7cf43021171e64156f70656#file-startup)

**Reply by worried-networking** [reply]
Thanks! Unfortunately, there are no messages I'm looking for :( I'm looking for something like this:
```
fa d4 | 19 00 | 00 00 00 00 | 02 | 06 | 0d 00 | ff | 00 00 00 | 12 34 56 78 | 34 31 39 31 37 31 37 31 31 30 38 30 31 | 12 34 | 2d df
```

Having `02` in byte 9 and one of the four following values in byte 10: `06` / `05` / `0b` / `0c`. That's the full initialisation sequence:
```
... | 02 06 | ...
... | 02 0b | ...
... | 02 0c | ...
... | 02 05 | ...
```

There might be occasional pings between those (messages with `00 06` and `00 0F` in those bytes), but that's fine. Feel free to just dump me everything you have without cherry-picking the data; I'll be processing it in Jupyter anyway :)

P.S. Oh yeah, don't tell me about the interface :D Even Tuya looks better than this abomination.

**Comment by worried-networking** [comment]
By the way, what does the physical switch on FSTWiFi do? Does it actually turn on/off the connected device? Are there any changes in the SupplyMaster app when you toggle this switch?

**Comment by RosemaryOrchard** [comment]
> By the way, what does the physical switch on FSTWiFi do? Does it actually turn on/off the connected device? Are there any changes in the SupplyMaster app when you toggle this switch?

It _appears_ to disable network control. So the switch itself still receives commands (I saw them go by in the logs, and heard the physical click), but nothing actually happened. In the app, I didn't see anything. Though as you noted the app is something from the deepest circles of hell 🤪 

Confusingly there's also a press button with a power icon on it that actually toggles the fan state.

Logs for the first 5 minutes after killing the power at the consumer unit and putting it back on (pulling the fuse didn't seem to do it), are here: https://gist.github.com/RosemaryOrchard/55925620a7cf43021171e64156f70656#file-startup-power-removed
At approximately 10/07/2022 15:21:11 power should have been restored, but there's plenty of logs before and after that so hopefully you find what you're after.

**Reply by worried-networking** [reply]
Yup, the messages I was looking for are here; thanks!

**Reply by RosemaryOrchard** [reply]
Phew! I was not relishing pulling power to the house again—though at least I got to test my "if there's a power outage, schedule the microwave to be restarted at 1am" automation 🤣

**Reply by worried-networking** [reply]
Well... I checked the logs, and I have no idea how to interpret them :D I see the difference in a few places, but it's not clear if those things can help distinct FSTWiFi from NTTWiFi, or if they're just different with each device :\

Just for the history books, here're two things I've noticed:

* In the initial `CODE_VERSION` message, my NTTWiFi has a string `4191717110801`, while in your logs, there's `4190117051001`. 
* In the `PING` messages my NTTWiFi has value `0x00 0x64 0x00` while yours FSTWiFi sends `0x00 0x56 0x00`.

At least it works without any changes 😅

**Reply by RosemaryOrchard** [reply]
> At least it works without any changes 😅

Yup. My extractor fan turned off when my dishwasher door automatically opened at the end of the cycle and then turned off again 10 minutes later! Success!!
